ZkSNARK/GeneralLemmas/MvXMul.lean

```diff
@@ -0,0 +1,196 @@
+import Mathbin.Data.MvPolynomial.Basic
+
+noncomputable section
+
+open Classical BigOperators
+
+open Set Function Finsupp AddMonoidAlgebra
+
+universe u v
+variable {R : Type u} 
+
+namespace MvPolynomial
+variable {σ : Type _} {e : ℕ} {n m : σ} {s : σ →₀ ℕ}
+
+section CommSemiringₓ
+
+variable [CommSemiringₓ R] {p q : MvPolynomial σ R}
+variable [CommSemiring R]
```

Comment:
Hmm, this should probably not be here -- we already have a `[CommSemiringₓ R]`. Adding this one
1. is not backward compatible with mathlib3
2. might cause a bunch of weird instance conflicts in the future

Reply:
They are proved using the `ring` tactics